source/standards/accounts-with-third-parties.html.md.erb

```diff
@@ -0,0 +1,146 @@
+---
+title: Managing accounts with third party services
+expires: 2018-04-01
+---
+
+# <%= current_page.data.title %>
+
+<%= partial :expires %>
+
+When developing a service, your team will often require the use of
+third-party services.  To access these services, you will need some
+sort of user account.  This page provides guidance on how to manage
+those accounts.
+
+## Terms
+
+In this page we distinguish between an account and an individual:
+
+- An *account* means an account with an identifier such as a username or
+  email address
+- An *individual* means a human being
+- A set of *credentials* means some sort of secret used to gain access
+  to a user account (for example: a password, a secret access key)
+
+## Goals and principles
+
+When managing credentials we have a few goals:
+
+- Ensure that the right people have access
+- Ensure that the wrong people do not have access
+  - Particularly: ensure when people leave the organization, their
+    access is revoked or expired
+- Identify who did what and when (audit)
+  - This goal has varying importance depending on context
+  - Technical limitations may prevent the goal from being achieved, or
+    only provide partial audit
+
+To meet these goals, we have some principles:
+
+- Prefer to minimise the number of secrets that an individual needs to
+  keep
+  - This helps with revocation because there are fewer sets of
+    credentials to revoke when someone leaves
+  - Passwords are particularly problematic secrets. Other forms of
+    secrets (eg secret access keys or OAuth tokens) are more easily
+    revoked and recycled.
```

Comment:
I don't think I agree that passwords are less easily revoked than secret access keys or OAuth tokens. Could you explain your reasoning?

I'm also not clear what you mean by recycled - do you mean replaced/rotated? To me, in this context, recycled would mean reuse of the same credentials for multiple accounts.

Reply:
good shout, I'll use "rotated"

Reply:
regarding "passwords are less easily revoked": I think I meant "passwords are less easily *rotated*" here.  I'll try to explain my reasoning:

The passwords I'm talking about here are used to identify individuals with third party services (not eg database passwords).  Other credentials like API keys are not supposed to be memorized; they are stored somewhere (eg `~/.aws/credentials`).  Rotating the credentials just means updating the places where they are stored.

Passwords are different.  Not everyone uses password managers, and those who do don't necessarily use them for every account.  If a password is memorized, cognitive effort has to go in to memorizing the new password.

Passwords are also globally unique per account, so if they are stored on a device (say, in a non-cloudy password manager like `pass`) then rotating the password involves work to get the new password to every device that has it stored.  This is different to API keys, where a separate key per device is often common, and so rotating a key involves only updating one device.